src/JsonSchema/Generator/NormalizerGenerator.php

```diff
@@ -69,14 +69,14 @@ public function generate(Schema $schema, string $className, Context $context)
             $methods[] = $this->createDenormalizeMethod($modelFqdn, $context, $class);
             $methods[] = $this->createNormalizeMethod($modelFqdn, $context, $class);
 
-            if ($this->useCacheableSupportsMethod) {
+            if ($this->canUseCacheableSupportsMethod()) {
```

Comment:
No need for that, you should set the private property of this class with the condition on the constructor